internal/controller/rotators/aws_common.go

```diff
@@ -0,0 +1,204 @@
+const (
+	// awsCredentialsKey is the key used to store AWS credentials in Kubernetes secrets.
+	awsCredentialsKey = "credentials"
+	// awsSessionNameFormat is the format string for AWS session names.
+	awsSessionNameFormat = "ai-gateway-%s"
+)
+
+// defaultAWSConfig returns an AWS config with adaptive retry mode enabled.
+// This ensures better handling of transient API failures and rate limiting.
+func defaultAWSConfig(ctx context.Context) (aws.Config, error) {
+	return config.LoadDefaultConfig(ctx,
+		config.WithRetryMode(aws.RetryModeAdaptive),
+	)
+}
+
+// STSClient defines the interface for AWS STS operations required by the rotators.
+// This interface encapsulates the STS API operations needed for OIDC token exchange
+// and role assumption.
+type STSClient interface {
+	// AssumeRoleWithWebIdentity exchanges a web identity token for temporary AWS credentials.
+	AssumeRoleWithWebIdentity(ctx context.Context, params *sts.AssumeRoleWithWebIdentityInput, optFns ...func(*sts.Options)) (*sts.AssumeRoleWithWebIdentityOutput, error)
+}
+
+// stsClient implements the STSOperations interface using the AWS SDK v2.
+// It provides a concrete implementation for STS operations using the official AWS SDK.
+type stsClient struct {
+	client *sts.Client
+}
+
+// NewSTSClient creates a new STSClient with the given AWS config.
+// The client is configured with the provided AWS configuration, which should
+// include appropriate credentials and region settings.
+func NewSTSClient(cfg aws.Config) STSClient {
+	return &stsClient{
+		client: sts.NewFromConfig(cfg),
+	}
+}
+
+// AssumeRoleWithWebIdentity implements the STSOperations interface by exchanging
+// a web identity token for temporary AWS credentials.
+//
+// This implements [STSClient.AssumeRoleWithWebIdentity].
+func (c *stsClient) AssumeRoleWithWebIdentity(ctx context.Context, params *sts.AssumeRoleWithWebIdentityInput, optFns ...func(*sts.Options)) (*sts.AssumeRoleWithWebIdentityOutput, error) {
+	return c.client.AssumeRoleWithWebIdentity(ctx, params, optFns...)
+}
+
+// awsCredentials represents a single set of AWS credentials, including optional
+// session token and region configuration. It maps to a single profile in an
+// AWS credentials file.
+type awsCredentials struct {
+	// profile is the name of the credentials profile.
+	profile string
+	// accessKeyID is the AWS access key ID.
+	accessKeyID string
+	// secretAccessKey is the AWS secret access key.
+	secretAccessKey string
+	// sessionToken is the optional AWS session token for temporary credentials.
+	sessionToken string
+	// region is the optional AWS region for the profile.
+	region string
+}
+
+// awsCredentialsFile represents a complete AWS credentials file containing
+// multiple credential profiles. It provides a structured way to manage
+// multiple sets of AWS credentials.
+type awsCredentialsFile struct {
+	// profiles maps profile names to their respective credentials.
+	profiles map[string]*awsCredentials
+}
```

Comment:
i don't think you need this. it should be possible to simply use `[] awsCredentials`. Then you don't need the sorting stuff below in the first place